### PR TITLE
Emulated v406 usio

### PR DESCRIPTION
**Pull request by clienthax**
Used by S357/variants arcade games.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 1]
```diff
@@ -226,6 +228,12 @@ usb_handler_thread::usb_handler_thread()
 
 		// DVB-T
 		check_device(0x1415, 0x0003, 0x0003, " PlayTV SCEH-0036");
+
+		// V406 USIO
+		if (check_device(0x0B9A, 0x0910, 0x0910, "USIO PCB rev00")) {
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp:233]:
```suggestion
		if (check_device(0x0B9A, 0x0910, 0x0910, "USIO PCB rev00"))
		{
```

File: rpcs3/Emu/Io/usio.cpp [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+		case 0x005A:
+		case 0x006A:
+		case 0x007A:
+		{
+			usio_log.notice("SetHopperRequest(Hopper: %d, Limit: 0x%04X)", (reg - 0x4A) / 0x10, get_u16("SetHopperLimit"));
+			break;
+		}
+		default:
+		{
+			//usio_log.error("Unhandled channel 0 register write: 0x%04X", reg);
```
Comment (Megamouse) [inline rpcs3/Emu/Io/usio.cpp:206]: Missing brek

File: rpcs3/Emu/Io/usio.cpp [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+		{
+			ensure(size == 0x70);
+			// NBGI2;USIO01;Ver1.00;JPN,Multipurpose with PPG.
+			q_replies.push({ 0x4E, 0x42, 0x47, 0x49, 0x32, 0x3B, 0x55, 0x53, 0x49, 0x4F, 0x30, 0x31, 0x3B, 0x56, 0x65, 0x72, 0x31, 0x2E, 0x30, 0x30, 0x3B, 0x4A, 0x50, 0x4E, 0x2C, 0x4D, 0x75, 0x6C, 0x74, 0x69, 0x70, 0x75, 0x72, 0x70, 0x6F, 0x73, 0x65, 0x20, 0x77, 0x69, 0x74, 0x68, 0x20, 0x50, 0x50, 0x47, 0x2E, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 });
+			q_replies.push({ 0x01, 0x00, 0x13, 0x00, 0x30, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x03, 0x02, 0x00, 0x08, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03, 0x00, 0x03, 0x00, 0x08, 0xE2, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 });
+			break;
+		}
+		default:
+		{
+			usio_log.fatal("Unhandled channel 0 register read: 0x%04X", reg);
```
Comment (Megamouse) [inline rpcs3/Emu/Io/usio.cpp:298]: Missing brek

File: rpcs3/Emu/Io/usio.cpp [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+#include "Input/pad_thread.h"
+#include "Emu/System.h"
+
+LOG_CHANNEL(usio_log);
+
+usb_device_usio::usb_device_usio()
+{
+	device = UsbDescriptorNode(USB_DESCRIPTOR_DEVICE, UsbDeviceDescriptor{
+														  .bcdUSB = 0x0110,
+														  .bDeviceClass = 0xff,
```
Comment (Megamouse) [inline rpcs3/Emu/Io/usio.cpp:16]: Why is everything here so far indented

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 5]
```diff
@@ -226,6 +228,13 @@ usb_handler_thread::usb_handler_thread()
 
 		// DVB-T
 		check_device(0x1415, 0x0003, 0x0003, " PlayTV SCEH-0036");
+
+		// V406 USIO
+		if (check_device(0x0B9A, 0x0910, 0x0910, "USIO PCB rev00"))
+		{
+			found_usio = true;
+		}
+
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp:237]: remove da empty line

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp [hunk 6]
```diff
@@ -648,6 +663,12 @@ error_code sys_usbd_register_ldd(ppu_thread& ppu, u32 handle, vm::ptr<char> s_pr
 		sys_usbd.warning("sys_usbd_register_ldd(handle=0x%x, s_product=%s, slen_product=0x%x) -> Redirecting to sys_usbd_register_extra_ldd", handle, s_product, slen_product);
 		sys_usbd_register_extra_ldd(ppu, handle, s_product, slen_product, 0x0B9A, 0x0800, 0x0800);
 	}
+	if (strcmp(s_product.get_ptr(), "PS3A-USJ") == 0)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_usbd.cpp:666]:
```suggestion
	else if (strcmp(s_product.get_ptr(), "PS3A-USJ") == 0)
```

File: rpcs3/Emu/Cell/lv2/sys_usbd.cpp (hunk 5 above)
Comment (clienthax) [reply rpcs3/Emu/Cell/lv2/sys_usbd.cpp:237]: Empty line wants to live!

File: rpcs3/Emu/Io/usio.cpp [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+	default:
+		// Follow to default emulated handler
+		usb_device_emulated::control_transfer(bmRequestType, bRequest, wValue, wIndex, wLength, buf_size, buf, transfer);
+		break;
+	}
+}
+
+void usb_device_usio::translate_input()
+{
+	const auto handler = pad::get_current_handler();
```
Comment (Megamouse) [inline rpcs3/Emu/Io/usio.cpp:91]: There should be a mutex here to ensure that the pad data is not messed with while we're trying to fetch it.
The same seems to be missing in Turntable.cpp

```suggestion
	std::lock_guard lock(pad::g_pad_mutex);
	const auto handler = pad::get_current_handler();
```

**Pull request merged**